src/parser/ad_parser.js

```diff
@@ -232,47 +232,62 @@ function parseWrapper(wrapperElement) {
  */
 function parseExtensions(collection, extensions) {
   extensions.forEach(extNode => {
-    const ext = new AdExtension();
-    const extNodeAttrs = extNode.attributes;
-    const childNodes = extNode.childNodes;
+    const ext = parseExtension(extNode);
 
-    if (extNode.attributes) {
-      for (let extNodeAttrKey in extNodeAttrs) {
-        const extNodeAttr = extNodeAttrs[extNodeAttrKey];
-
-        if (extNodeAttr.nodeName && extNodeAttr.nodeValue) {
-          ext.attributes[extNodeAttr.nodeName] = extNodeAttr.nodeValue;
-        }
-      }
+    if (ext) {
+      collection.push(ext);
     }
+  });
+}
 
-    for (let childNodeKey in childNodes) {
-      const childNode = childNodes[childNodeKey];
-      const txt = parserUtils.parseNodeText(childNode);
-
-      // ignore comments / empty value
-      if (childNode.nodeName !== '#comment' && txt !== '') {
-        const extChild = new AdExtensionChild();
-        extChild.name = childNode.nodeName;
-        extChild.value = txt;
-
-        if (childNode.attributes) {
-          const childNodeAttributes = childNode.attributes;
-
-          for (let extChildNodeAttrKey in childNodeAttributes) {
-            const extChildNodeAttr = childNodeAttributes[extChildNodeAttrKey];
+/**
+ * Parses an extension child node
+ * @param {Object} extNode - The extension node to parse
+ * @param {Boolean} isChild - A flag to tell how to parse the node
+ * @return {Object}
+ */
+function parseExtension(extNode, isChild = false) {
+  // Ignore comments
+  if (extNode.nodeName === '#comment') return null;
+
+  let ext;
+  const extNodeAttrs = extNode.attributes;
+  const childNodes = extNode.childNodes;
+
+  // Only children have a name and value
+  if (isChild) {
+    ext = new AdExtensionChild();
+    const txt = parserUtils.parseNodeText(extNode);
+
+    if (txt !== '') {
+      ext.value = txt;
```

Comment:
if there are many nested nodes, like: 
```
<Extension type="FoodsExtension">
  <Foods>
    <Vegetables>
      <Vegetable>Carrot</Vegetable>
      <Vegetable>Beets</Vegetable>
    </Vegetables>
    <Fruits>
      <Fruit>Apple</Fruit>
      <Fruit>Strawberry</Fruit>
    </Fruits>
  </Foods>
</Extension>
```
Node `<Foods>` will have value: `"value": "Carrot\n                        Beets\n                    \n                    \n                        Apple\n                        Strawberry",`.
So maybe we could try to find if that node contains valuable `value`, like `#text` or `#cdata-section`.